pallets/automation-time/src/lib.rs

```diff
@@ -1075,7 +1142,8 @@ pub mod pallet {
 										ScheduledTasksOf::<T> {
 											tasks: account_task_ids,
 											weight: weight.saturating_sub(
-												task.action.execution_weight::<T>() as u128,
+												task.action.execution_weight::<T>().unwrap_or(0)
```

Comment:
My thought here is that if we cannot decode the call then the safest thing to do is keep the weight as is. In order to make this better we would need to save the weight we got initially onto the task and use it here